src/components/CustomTableHead/CustomTableHead.tsx

```diff
@@ -0,0 +1,111 @@
+import { Box, Checkbox, TableCell, TableHead, TableRow, TableSortLabel } from '@mui/material';
+import { ChangeEvent, MouseEvent } from 'react';
+import { CustomTableProps } from '../CustomTable/CustomTable.types';
+import { SortingDirection } from '../../common/utils/SortingDirection';
+import { t } from 'i18next';
+
+export function CustomTableHead<T extends Record<PropertyKey, unknown>>(props: CustomTableProps<T>) {
+  const { columnConfig, selectedRowsIds, rows, setColumnConfig, setSelectedRowsIds, tableName } = props;
+
+  const isSelected = (row: T) => {
+    return (
+      selectedRowsIds.filter((item) => {
+        return (row.id as string) === item;
+      }).length > 0
+    );
+  };
+
+  const createSortHandler = (property: keyof T) => {
+    return (event: MouseEvent<unknown>) => {
+      handleRequestSort(event, property);
+    };
+  };
+
+  function updateSortingRule(index: number, columnName: string, sortDirection: SortingDirection) {
+    const updatedConfig = [...columnConfig];
+    updatedConfig.splice(index, 1, { id: columnName as keyof T, label: columnName.toUpperCase(), sortDirection: sortDirection });
+    setColumnConfig(updatedConfig);
+  }
+
+  const handleSelectAllClick = (event: ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      //check whether the rows have been selected
+      const newSelectedRowsIds: readonly string[] = rows
+        .map((row, index) => {
+          return !isSelected(row) ? String(index) : String(-1);
+        })
```

Comment:
Why you are converting index into string ?

Reply:
Removed, now using row.id in the latest version.